pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java

```diff
@@ -224,6 +225,15 @@ public void removeProducer(Producer producer) {
         return  CompletableFuture.completedFuture(null);
     }
 
+    @Override
+    public CompletableFuture<Consumer> subscribe(SubscriptionOption option) {
+        return subscribe(option.getCnx(), option.getSubscriptionName(), option.getConsumerId(), option.getSubType(),
+                option.getPriorityLevel(), option.getConsumerName(), option.isDurable(), option.getStartMessageId(),
+                option.getMetadata(), option.isReadCompacted(), option.getInitialPosition(),
+                option.getStartMessageRollbackDurationSec(), option.isReplicatedSubscriptionStateArg(),
+                option.getKeySharedMeta());
+    }
+
     @Override
     public CompletableFuture<Consumer> subscribe(final TransportCnx cnx, String subscriptionName, long consumerId,
```

Comment:
This interface has been marked as `Deprecated`, we'd better change it to `private CompletableFuture<Consumer> internalSubscribe` and delete `@Override` to avoid confusing.